DEPRECATIONS.md

```diff
@@ -0,0 +1,67 @@
+# Deprecations & Breaking Changes in KEDA
+
+This document describes how KEDA handles deprecations and breaking changes and should provide clarity to KEDA end-users on what to expect when upgrading to a new version of KEDA.
+
+If you are interested in the current open deprecations, we recommend checking our [GitHub Discussions](https://github.com/kedacore/keda/discussions/categories/deprecations) or the open breaking changes](https://github.com/kedacore/keda/issues?q=is%3Aopen+label%3Abreaking-change+sort%3Aupdated-desc).
+
+## Our versioning strategy
+
+KEDA follows the [Semantic Versioning](https://semver.org/) strategy and can only introduce breaking changes in new major versions.
+
+However, depending on the nature of the change, we may introduce deprecations in existing minor versions. These changes always must be backwards-compatible.
+
+Depending on the area of the change, we might not need to wait for the next major version of KEDA to introduce a breaking change.
+
+### KEDA runtime
+
+The KEDA runtime, which consists of the operator and metric server, is not allowed to make breaking changes unless the major version is incremented.
+
+There are no exceptions to this rule.
+
+### Custom Resource Definitions (CRDs)
+
+Our customer resource definitions (CRDs) have a seperate versioning from the KEDA runtime given they have an `apiVersion`.
+
+KEDA is allowed to make breaking changes to the CRDs, when a new `apiVersion` is introduced.
+
+However, the KEDA runtime must support all current API versions until the next major version of KEDA.
+
+### Autoscaling triggers and its metadata
+
+Until KEDA introduces a [versioning scheme for autoscaling triggers](https://github.com/kedacore/keda/issues/613), it is not allowed to make breaking changes for triggers.
+
+The only exception to this rule is when a new `apiVersion` is introduced.
```

Comment:
IMO we shouldn't maintain more than 1 version, I don't have any problem keeping old versions in the code till next major KEDA versions, but we should define the if we are going to maintain multiple versions or only 1-2.

My vote is for maintaining only the latest version, once we create a new version, the previous one enters automatically in deprecated state in X KEDA versions (ore even more, deprecated versions could be deleted e.g, after 2 KEDA versions). But as I said in other comments, alpha/beta means other thing, doing this we are treating as stable one api which isn't stable yet

Reply:
I think https://github.com/kedacore/governance/pull/70#discussion_r968071392 also covers this discussion.

Reply:
Agree with @JorTurFer on this.

Reply:
Also, `v1alpha1` `apiVersion` allows us to do any change whenever we want.

Reply:
@zroubalik see https://github.com/kedacore/governance/pull/70#discussion_r968071392 but it's not because you can that you should.

Reply:
As we commented in the last standup, I wouldn't keep deprecated fields/behaviours related with CRDs (including the scaler behaviour) for more than X KEDA versions (for me 2). Once we have introduced the scaler versioning, we can discuss how we maintain the each scaler version, but ATM I prefer to maintain them only 2 KEDA versions. 
2 KEDA versions means 6 months approx to address the changes, I think that it's enough time to do it. 
As we discussed, too, I'd add a clear deprecation log saying that the feature will be removed in version XXXX (N+2).

Reply:
That sounds good to me; but I really think we should respect apiVersions. So are you suggesting that API versions will get bumped after 2 versions? Because solely relying on KEDA version and ignoring the CRD version for this is not correct imo.

Reply:
I'm not totally sure if I understand your point. I suggest removing every CRD related deprecation 2 version later. If we deprecate spec.XXXX in version 2.9, that deprecated code will be removed in version 2.11. 

I understand your concerns about semantic versioning, but Kubernetes itself doesn't follow it and we are creating a project on top of Kubernetes. E.g: Kubernetes deprecated autoscalingv2/beta2 in v1.23, and they will remove it in v1.26, the major version is 1, and they are removing the whole API, not only a field (K8s does it too in case of simple fields).

In general, I agree that we should follow semver for KEDA itself, but for everything related with CRDs and K8s I wouldn't do it, if we are on top of k8s, we should follow k8s approach

Reply:
Updated after our discussion